### PR TITLE
scripts: Enable testing script on Windows

### PR DESCRIPTION
**Pull request by juan-lunarg**
Currently can't run on GHA due to security permissions.

It works locally on my Windows machine now though.

### CODE REVIEW

File: scripts/github_ci_win_linux.py [hunk 1]
```diff
@@ -16,14 +16,21 @@
...
 def Build(args):
     config = args.configuration
 
+    # Since this script uses Ninja to build Windows users need to be in a developer command prompt.
+    if common_ci.IsWindows():
+        # This environment variable is arbitrary. I just picked one set by the developer command prompt.
+        if "VSCMD_ARG_TGT_ARCH" not in os.environ:
+            print("This script must be invoked in a developer command prompt!")
+            sys.exit(1)
+
```
Comment (juan-lunarg) [inline scripts/github_ci_win_linux.py:33]: This is my attempt at checking for the developer command prompt being present. Not super happy with this approach but it works.

Tested with a regular powershell prompt and it gave the proper error message.

File: scripts/common_ci.py [hunk 2]
```diff
@@ -210,21 +210,28 @@ def BuildProfileLayer():
...
-    lvt_env['LD_LIBRARY_PATH'] = os.path.join(TEST_INSTALL_DIR, 'lib')
-    lvt_env['VK_LAYER_PATH'] = os.path.join(TEST_INSTALL_DIR, 'share/vulkan/explicit_layer.d')
-    lvt_env['VK_DRIVER_FILES'] = os.path.join(TEST_INSTALL_DIR, 'share/vulkan/icd.d/VkICD_mock_icd.json')
+    # defined by GNUInstallDirs. This makes setting VK_LAYER_PATH and other environment variables trivial/robust.
+    if IsWindows():
+        # TODO: Enable Windows testing on GHA. GHA run the test with elevated permissions which causes the loader
+        # to ignore VK_LAYER_PATH for security reasons.
+        #
+        # Perhaps this can be fixed with GHA or more likely with registry editing.
+        # Ideally the registry editing is only done during GHA to avoid impacting local users who run this script.
```
Comment (juan-lunarg) [inline scripts/common_ci.py:224]: Sadly I'm running into a security issue with running this script on github actions.

`Loader is running with elevated permissions. Environment variable VK_LAYER_PATH will be ignored`

@charles-lunarg has suggested editing the registry. Which may be what I have to in another PR unless anyone else has any ideas.

**Review by juan-lunarg — COMMENTED** [review]
I'm now able to run the script locally on my Windows machine.

File: scripts/common_ci.py (hunk 2 above)
Comment (charles-lunarg) [reply scripts/common_ci.py:224]: USE_UNSAFE_FILE_SEARCH is a build option you can use to allow VK_LAYER_PATH to be used while in admin mode.

Comment (juan-lunarg) [reply scripts/common_ci.py:224]: I'll try that in a follow up PR.

File: scripts/github_ci_win_linux.py (hunk 1 above)
Comment (artem-lunarg) [reply scripts/github_ci_win_linux.py:33]: Yep, detecting anything related to VS was always a pain. Not sure if people do this but one option is to check if you can run cl.exe process but also can not recommend this as a good option, just something that came to mind.

**Review by artem-lunarg — APPROVED** [review]
LGTM 🙏

**Review by artem-lunarg — APPROVED** [review]
LGTM 🙏

File: scripts/github_ci_win_linux.py (hunk 1 above)
Comment (juan-lunarg) [reply scripts/github_ci_win_linux.py:33]: Yeah I also thought about checking for cl.exe proccess in the path. But I didn't see too much more merit to that approach over checking an environment variable. I guess we will see how this approach goes for now. I can change if it ever becomes a problem.
